pkgs/applications/audio/surge/default.nix

```
@@ -0,0 +1,51 @@
+{ stdenv, fetchFromGitHub, premake5, pkgconfig, cmake
+,cairo, libxkbcommon, libxcb, xcb-util-cursor, xcbutilkeysyms, ncurses, which, getopt
+,python, gnome3, lato }:
+
+stdenv.mkDerivation rec {
+  pname = "surge";
+  version = "1.6.6";
+
+  src = fetchFromGitHub {
+    owner = "surge-synthesizer";
+    repo = pname;
+    rev = "release_${version}";
+    sha256 = "0al021f516ybhnp3lhqx8i6c6hpfaw3gqfwwxx3lx3hh4b8kjfjw";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ premake5 pkgconfig cmake ncurses which getopt python ];
+
+  buildInputs = [  cairo libxkbcommon libxcb xcb-util-cursor xcbutilkeysyms gnome3.zenity lato ];
+
+  buildFlags = [ "config=release_x64" ];
+
+  enableParallelBuilding = true;
+
+  patchPhase = ''
+    patchShebangs build-linux.sh
+    substituteInPlace src/common/SurgeStorage.cpp --replace "/usr/share/Surge" "$out/share/surge"
+  '';
+
+  configurePhase = ''
+      ./build-linux.sh premake
+      python scripts/linux/emit-vector-piggy.py .
+    '';
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2
+    cp -r target/lv2/Release/Surge.lv2/ $out/lib/lv2
+    mkdir -p $out/lib/vst
+    cp target/vst3/Release/Surge.so $out/lib/vst
+    mkdir -p $out/share/surge
+    cp -r resources/data/* $out/share/surge/
+  '';
+
```

```suggestion

  doInstallCheck = true;

  installCheckPhase = ''
    ./build-linux.sh build -p headless
    ./target/headless/Release/Surge/Surge-Headless
  '';
```

I made the tests work :) They're install checks because `surge-headless` looks for the resources in the install directory.

---

Yeah that makes sense - there’s no reason to run them any time other than after build. 

A couple of the tests have tolerance problems I’m chasing and will fail about one time in 20 just FYI.